portfolio/src/main/java/com/google/sps/servlets/ListCommentsServlet.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+// import com.google.sps.data.Task;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+// To List comments
+@WebServlet("/list-comments")
+public class ListCommentsServlet extends HttpServlet {
+
+    @Override
+    public void doGet (HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+        
+        ArrayList<Comment> comments = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            String name = (String) entity.getProperty("name");
+            String message = (String) entity.getProperty("message");
+            long timestamp = (long) entity.getProperty("timestamp");
+
+            Comment comment = new Comment (name, message, timestamp);
+            comments.add(comment);
+        }
```

Comment:
If you want an extra challenge, you can try some functional programming here: 

* https://winterbe.com/posts/2014/07/31/java8-stream-tutorial-examples/
* https://www.baeldung.com/java-iterable-to-stream

Reply:
I would also break this out to a function that takes `results` and returns a that `ArrayList<Comment>`